src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs

```diff
@@ -0,0 +1,127 @@
+
+namespace System.Numerics.Tensors
+{
+    public static partial class Tensor
+    {
+        public static Tensor<T> Create<T>(bool mustPin, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = mustPin ? GC.AllocateArray<T>((int)linearLength, mustPin) : (new T[linearLength]);
+            return new Tensor<T>(values, lengths.ToArray(), mustPin);
+        }
+
+        public static Tensor<T> Create<T>(bool mustPin, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = mustPin ? GC.AllocateArray<T>((int)linearLength, mustPin) : (new T[linearLength]);
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), mustPin);
+        }
+
+        public static Tensor<T> Create<T>(T[] values, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            if (linearLength != values.Length)
+                throw new Exception();
+
+            return new Tensor<T>(values, lengths.ToArray(), false);
+        }
+
+        public static Tensor<T> Create<T>(T[] values, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            if (linearLength != values.Length)
+                throw new Exception();
+
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), false);
+        }
+
+        //public static Tensor<T> Create(T* address, ReadOnlySpan<nint> lengths);
+        //public static Tensor<T> Create(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides);
+
+        public static Tensor<T> CreateUninitialized<T>(bool mustPin, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = GC.AllocateUninitializedArray<T>((int)linearLength, mustPin);
+            return new Tensor<T>(values, lengths.ToArray(), mustPin);
+        }
+
+        public static Tensor<T> CreateUninitialized<T>(bool mustPin, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = GC.AllocateUninitializedArray<T>((int)linearLength, mustPin);
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), mustPin);
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths) => new SpanND<T>(address, lengths, false);
```

Comment:
Why do we need CreateSpan APIs on Tensor? Can't someone just use the SpanND constructors?

Reply:
We should be able to.

@tannergooding do you remember why you had these static methods along with the constructors?

Reply:
Nope, it was probably just me being consistent across types or potentially due to the `static abstracts with interfaces`.